src/main/scala/com/sky/kafka/configurator/TopicConfigurationParser.scala

```diff
@@ -29,17 +31,18 @@ object TopicConfigurationParser extends AutoDerivation {
 
   implicit val stringMapDecoder: Decoder[Map[String, String]] = Decoder.instance { cursor =>
     def stringify(json: Json): Json = json.asNumber
-      .map(num => Json.fromString(num.truncateToInt.toString))
+      .flatMap(_.toInt)
+      .map(n => Json.fromString(n.toString))
       .getOrElse(json)
 
     def failWithMsg(msg: String) = DecodingFailure(msg, List.empty)
 
     for {
       jsonObj <- cursor.value.asObject.toRight(failWithMsg(s"${cursor.value} is not an object"))
-      valuesAsJsonStrings = jsonObj.withJsons(stringify).toMap
-      stringMap <- valuesAsJsonStrings
-        .mapValues(json => json.asString.toRight(failWithMsg(s"$json is not a string")))
-        .sequenceU
-    } yield stringMap
+      valuesAsJsonStrings = jsonObj.mapValues(stringify)
+      stringMap <- valuesAsJsonStrings.toList.traverse[Decoder.Result, (String, String)] {
```

Comment:
https://stackoverflow.com/questions/53945996/cats-effecthow-to-transform-mapx-ioy-to-iomapx-y